features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -0,0 +1,903 @@
+/*
+ * Copyright (c) 2017, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "AT_CellularNetwork.h"
+#include "nsapi_ppp.h"
+#include "CellularUtil.h"
+#include "CellularLog.h"
+
+using namespace std;
+using namespace mbed_cellular_util;
+using namespace mbed;
+
+AT_CellularNetwork::AT_CellularNetwork(ATHandler &atHandler) : AT_CellularBase(atHandler),
+    _stack(NULL), _uname(NULL), _pwd(NULL), _ip_stack_type_requested(DEFAULT_STACK), _ip_stack_type(DEFAULT_STACK), _cid(-1),
+    _op_act(operator_t::RAT_UNKNOWN), _authentication_type(CHAP), _last_reg_type(C_REG)
+{
+
+    _at.set_urc_handler("NO CARRIER", callback(this, &AT_CellularNetwork::urc_no_carrier));
+
+    memset(_apn, 0, MAX_ACCESSPOINT_NAME_LENGTH);
+
+#ifdef MBED_CONF_APP_CELLULAR_APN
```

Comment:
This is an (hypothetical) application config setting - you can't access that from here.

If you want a default APN setting JSON it needs to be in your own cellular mbed_lib.json.

Reply:
Changed APN initialization from CellularNetwork to application responsibility
https://github.com/AriParkkila/mbed-os/commit/31c4e400a5f9738f07cedba0ad71ad3cfd1a9381